js/imageeditor/shared/core/editor.ts

```diff
@@ -868,21 +878,36 @@ export class LayerManager {
 
 	reset_layers(width: number, height: number, persist = false): void {
 		const _layers_to_recreate = persist
-			? this.layers.map((layer) => layer.name)
-			: this.layer_options.layers;
+			? this.layers.map((layer) => [layer.name, layer.id])
+			: this.layer_options.layers.map((layer) => [layer, undefined]);
 
 		this.layers.forEach((layer) => {
 			this.delete_layer(layer.id);
 		});
 
-		for (const layer of _layers_to_recreate) {
+		let i = 0;
+		for (const [layer_name, layer_id] of _layers_to_recreate) {
 			this.create_layer({
 				width,
 				height,
-				layer_name: layer,
-				user_created: this.layer_options.layers.includes(layer) ? false : true
+				layer_name: layer_name,
+				user_created: this.layer_options.layers.find((l) => l === layer_name)
+					? false
+					: true,
+				layer_id: layer_id
 			});
+			i++;
 		}
+
+		if (!persist) {
+			this.active_layer = this.layers[0].container;
+			this.active_layer_id = this.layers[0].id;
+		}
```

Comment:
If we aren't persisting the current state (because we have completely refreshed all layers for example), will will default to the first layer.